src/client/editor/objects/Model.js

```diff
@@ -6,21 +6,39 @@ export default class Model extends THREE.Object3D {
     super();
     this.type = "Model";
 
-    this.model = null;
+    this._src = null;
     this.animations = [];
     this.clipActions = [];
-
     this._mixer = new THREE.AnimationMixer(this);
   }
 
-  setModel(object, animations) {
-    this.model = object;
+  get src() {
+    return this._src;
+  }
+
+  set src(value) {
+    this.load(value).catch(console.error);
+  }
+
+  loadGLTF(src) {
+    return new Promise((resolve, reject) => {
+      new THREE.GLTFLoader().load(src, resolve, null, reject);
+    });
+  }
```

Comment:
Hmm, a bit confusing to tell if `set src` and `loadGLTF` here are actually used. I'm pretty sure `loadGLTF` here never gets called, since we override it in ModelNode and we never call `Model.loadGLTF` directly, or indirectly.

Reply:
Neither are called. My only excuse is that Spoke objects are just Three.js Object3Ds. They should work correctly outside the editor and could be used in a future runtime project. Spoke's Nodes are tied to the editor and so they override those methods. There's not much code in those overridden methods so I don't see much harm in it. Ideally I'd like to see these objects split out into another library of useful ThreeJS objects.

Reply:
I would add some comments at least. It would help anyone else reading the spoke code.